### PR TITLE
Introduce Component and ComponentPackage types

### PR DESCRIPTION
**Pull request by jpogran-hashi**
## Related Issue

Fixes # 

## Description

This provides a structured way to handle component addresses
in the Terraform registry, enabling validation and error
reporting. The new types and functions facilitate the
installation workflow for components.

- Added Component and ComponentPackage structs to represent registry
  components and their addresses.
- Implemented ParseComponentSource function to parse and validate
  component registry addresses.
- Created unit tests for ParseComponentSource to ensure correct parsing
  and error handling.

## Rollback Plan

- [ ] If a change needs to be reverted, we will roll out an update to the code within 7 days.

## Changes to Security Controls

Are there any changes to security controls (access controls, encryption, logging) in this pull request? If so, explain.

### CODE REVIEW

**Review by mjyocca — APPROVED** [review]

File: component_test.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+			}
+
+			if diff := cmp.Diff(addr, test.want); diff != "" {
+				t.Errorf("wrong result\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestParseComponentSource(t *testing.T) {
```
Comment (mjyocca) [inline component_test.go:112]: Test cases look great! Not required, but wonder if we should include any additional edge cases. Such as including a protocol or query parameters.

File: component.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+		Package: ComponentPackage{
+			Host: host,
+		},
+		Subdir: subDir,
+	}
+
+	// Prevent potential parsing collisions with known VCS hosts
+	// These hostnames are reserved for direct VCS installation
+	if host == svchost.Hostname("github.com") || host == svchost.Hostname("bitbucket.org") || host == svchost.Hostname("gitlab.com") {
+		return ret, fmt.Errorf("can't use %q as a component registry host, because it's reserved for installing directly from version control repositories", host)
```
Comment (liamcervante) [inline component.go:113]: Is it worth leaving a comment here that this may change if Gitlab does start hosting components? Or do we think that is unlikely given Stacks position as a TFC-only feature?

File: component_test.go (hunk 1 above)
Comment (mjyocca) [reply component_test.go:112]: Added two edge case tests.

File: component.go (hunk 2 above)
Comment (jpogran-hashi) [reply component.go:116]: @liamcervante I've updated the block to have a comment explaining this may change in the future

**Review by liamcervante — APPROVED** [review]
